airbyte-integrations/connectors/destination-weaviate/integration_tests/spec.json

```diff
@@ -0,0 +1,285 @@
+            "title": "OpenAI",
+            "type": "object",
+            "properties": {
+              "mode": {
+                "title": "Mode",
+                "default": "openai",
+                "const": "openai",
+                "enum": ["openai"],
+                "type": "string"
+              },
+              "openai_key": {
+                "title": "OpenAI API key",
+                "airbyte_secret": true,
+                "type": "string"
+              }
+            },
+            "required": ["openai_key"],
+            "description": "Use the OpenAI API to embed text. This option is using the text-embedding-ada-002 model with 1536 embedding dimensions."
+          },
+          {
+            "title": "Cohere",
+            "type": "object",
+            "properties": {
+              "mode": {
+                "title": "Mode",
+                "default": "cohere",
+                "const": "cohere",
+                "enum": ["cohere"],
+                "type": "string"
+              },
+              "cohere_key": {
+                "title": "Cohere API key",
+                "airbyte_secret": true,
+                "type": "string"
+              }
+            },
+            "required": ["cohere_key"],
+            "description": "Use the Cohere API to embed text."
+          },
+          {
+            "title": "From Field",
+            "type": "object",
+            "properties": {
+              "mode": {
+                "title": "Mode",
+                "default": "from_field",
+                "const": "from_field",
+                "enum": ["from_field"],
+                "type": "string"
+              },
+              "field_name": {
+                "title": "Field name",
+                "description": "Name of the field in the record that contains the embedding",
+                "examples": ["embedding", "vector"],
+                "type": "string"
+              },
+              "dimensions": {
+                "title": "Embedding dimensions",
+                "description": "The number of dimensions the embedding model is generating",
+                "examples": [1536, 384],
+                "type": "integer"
+              }
+            },
+            "required": ["field_name", "dimensions"],
+            "description": "Use a field in the record as the embedding. This is useful if you already have an embedding for your data and want to store it in the vector store."
+          },
+          {
+            "title": "Weaviate vectorizer",
+            "type": "object",
+            "properties": {
+              "mode": {
+                "title": "Mode",
+                "default": "no_embedding",
+                "const": "no_embedding",
+                "enum": ["no_embedding"],
+                "type": "string"
+              }
+            },
+            "description": "Do not calculate embeddings (suitable for classes with configured vectorizers to calculate embeddings within Weaviate)"
```

Comment:
I didn't see this but maybe I missed it:

Should we add some validation in check that prevents you from running with `no_embedding` if it's not properly configured in the weaviate side?

Reply:
Good point, however it might be a valid use case if a user doesn't want to use vector search but only "regular" full text search. I changed to "No embedding" and mention the vectorizer in the description.